types/types.go

```diff
@@ -30,7 +30,7 @@ type ImageSource interface {
 	// It may use a remote (= slow) service.
 	GetManifest([]string) ([]byte, string, error)
 	// Note: Calling GetBlob() may have ordering dependencies WRT other methods of this type. FIXME: How does this work with (docker save) on stdin?
-	GetBlob(digest string) (io.ReadCloser, error)
+	GetBlob(digest string) (io.ReadCloser, int64, error)
```

Comment:
Should document the return value.  Also, _if_ the value is defined to be optional, return 0 if unknown, this should be documented here.  (Alternatively, it might be a mandatory return value, and then the Docker client would fail if the header is missing or invalid.)